lib/hubspot/oauth.rb

```diff
@@ -0,0 +1,46 @@
+require 'httparty'
+module Hubspot
+  class OAuth < Connection
+    include HTTParty
+    DEFAULT_OAUTH_HEADERS = {"Content-Type" => "application/x-www-form-urlencoded;charset=utf-8"}
+    class << self
+      def refresh(params)
+        params.stringify_keys!
+        no_parse = params.delete("no_parse") { false }
+        body = {
+          client_id: params["client_id"] || Hubspot::Config.client_id,
+          grant_type: "refresh_token",
+          client_secret: params["client_secret"] || Hubspot::Config.client_secret,
+          redirect_uri: params["redirect_uri"] || Hubspot::Config.redirect_uri,
+          refresh_token: params["refresh_token"]
+        }
+        response = post(oauth_url, body: body, headers: DEFAULT_OAUTH_HEADERS)
+        log_request_and_response oauth_url, response, body
+        raise(Hubspot::RequestError.new(response)) unless response.success?
+
+        no_parse ? response : response.parsed_response
+      end
+
+      def create(params)
```

Comment:
Are the implementations of `create()` and `refresh()` the same here? If so, can these be moved to a single function so there aren't two separate implementations?

Reply:
@cbisnett they aren't quite the same. Notice the `grant_type` param is different ("refresh_token" vs. "authorization_code"), and also the last param is different. It's `refresh_token` in the "refresh" method, but it's `code` in the "create" method. 
But yeah they're pretty similar, and if you wanted to refactor out the common components, that would probably be good.